usr/local/www/interfaces_vlan_edit.php

```diff
@@ -80,7 +80,7 @@
 
 	do_input_validation($_POST, $reqdfields, $reqdfieldsn, $input_errors);
 
-	if ($_POST['tag'] && (!is_numericint($_POST['tag']) || ($_POST['tag'] < '1') || ($_POST['tag'] > '4094'))) {
+	if (!is_numericint($_POST['tag']) || ($_POST['tag'] < '1') || ($_POST['tag'] > '4094')) {
```

Comment:
This is the issue here - when the user puts VLAN tag "0" it gets $_POST['tag'] false - and so the "if" statement never gets even executed. The code thinks that "tag" was false, but actually the value "0" was supplied and gets written to the config.
Because this parameter is required anyway, we can skip $_POST['tag'] boolean check - it has to be numericint from 1 to 4094 in all cases.